<a href="https://colab.research.google.com/github/A109010109/Computational-Thinking-and-Program-Design/blob/main/%E6%9D%8E%E7%8F%88%E5%A6%82_Week10_%E4%BD%9C%E6%A5%AD.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 基本程式設計(應用篇)

### 4-2 運算思維四個基石

#### 4-2-4  問題拆解 - 打造你的聊天機器人

「哈囉，大家好。我叫Zyxo64，是一個聊天機器人。

在嗜好與興趣方面，我很喜歡美食，也喜歡養動物。

很高興有機會來到世新大學，請大家多多指教。

嗨！妳好，請問大名？：Francis。

Francis，很高興認識你。

In [2]:
# 1.聊天機器人自我介紹
print("嗨。我叫Penomeco，是一個聊天機器人。")
first = "我很喜歡音樂，也喜歡Fanxy Child"
print( first )


print("到時候是西元", int(year) + nyears + 1911, "年") # 將今年年份轉換
為整數
print("也就是民國", int(year) + nyears, "年") # 民國幾年

# 2. 打招呼，取得對方名字
name = input("嗨。名字是？ ")
print("你好", name, "반갑워요")

嗨。我叫Penomeco，是一個聊天機器人。
我很喜歡音樂，也喜歡Fanxy Child
請問一下今年是西元2021年，那是民國幾年呢？ 110
對！應該沒錯，謝謝妳！
Francis，猜猜看我今年幾歲？ 20
哇！你好聰明，我今年的確 20
我再 80 年就滿100歲了。
到時候是西元 2101 年


NameError: ignored

抱歉，請問一下今年是西元2021年，那是民國幾年呢？：*110*。

對！沒錯，謝謝妳！

**Francis**，猜猜看我今年幾歲？ ：**18**。

哇！你好聰明，我今年的確*18*。

再82年我就100歲了。

那時候是西元 2103年，也就是民國192年囉！

In [4]:
# 3. 數學運算
## 3-1 從使用者取得今年年份
year = input("今年是西元2021年，是民國幾年？ ")
print("嗯。")

## 3-2 請使用者猜年齡
nage = int(input("猜我今年幾歲"))
print("的確", nage)

## 3-3 計算聊天機器人滿100歲的年份
hyears = 100 - nage
print("我再", hyears, "年就滿100歲了。")


今年是西元2021年，是民國幾年？ 110
嗯。
猜我今年幾歲20
的確 20
我再 80 年就滿100歲了。


我喜歡吃巧克力，也喜歡嘗試各種新食物。

你呢？你最喜歡的食物是什麼？**:披薩**。

我也喜歡 *披薩*。

你多久吃一次 *披薩*？**：每天**。

真有趣。不知道這樣對健康好不好。

In [5]:
# 4 興趣
## 4-1 食物話題
print("我喜歡巧克力，也喜歡anything new。")
food = input("what bout you？： ")
print("我也喜歡", food)
question = "how often" + food + "?: "
howoften = input(question)
print("interesting")

我喜歡巧克力，也喜歡anything new。
what bout you？： tea
我也喜歡 tea
how oftentea?: 3
interesting


我最喜歡的動物是長頸鹿。你呢？ **：烏龜**。

我不喜歡*烏龜*！

不知道 *烏龜* 喜不喜歡吃 *披薩*？

In [6]:
# 4 興趣
## 4-2 動物話題
animal = input("我最喜歡貓。； ")
print("我喜歡", animal)
print("不知道", animal, "能次能夠喝", food, "?")

我最喜歡貓。； 恩
我喜歡 恩
不知道 恩 能次能夠吃 tea ?


你今天的心情如何？ **：很開心**。

可以告訴我你 *開心* 的原因嗎？**因為週末了！** 

祝你週末愉快！

我喜歡跟你聊天，現在有點累了，改天再聊吧！！

再見 *Francis*！

In [7]:
# 5. 關於心情的對話
feeling = input("what do you do； ")
print("can you tell me", feeling, "的原因。 ")
reason = input("為什麼： ")
print("thanks")

what do you do； no
can you tell me no 的原因。 
為什麼： ji
thanks


我喜歡跟你聊天，現在有點累了，改天再聊吧！！

再見 *Francis*！

In [14]:
# 6. 道別
print("週末愉快。")
print("下次再聊。")
name ="penomeco "
print("再見", name, "我喜歡跟你聊天！")

週末愉快。
下次再聊。
再見 penomeco  我喜歡跟你聊天！


## 期末報告

### 組長(學號姓名):



### 組員(學號姓名):



### 題目:

### 百字腳本

